pkg/injector/bootstrap.go

```diff
@@ -140,25 +140,28 @@ func (b *BootstrapSecretRotator) getBootstrapSecrets() []*corev1.Secret {
 func (b *BootstrapSecretRotator) rotateBootstrapSecrets(ctx context.Context) {
 	bootstrapSecrets := b.getBootstrapSecrets()
 	for _, secret := range bootstrapSecrets {
-		bootstrapCert, err := getCertFromSecret(secret)
-		if err != nil {
-			log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrObtainingCertFromSecret)).
-				Msgf("Error getting cert %s from secret %s/%s", bootstrapCert, secret.Namespace, secret.Name)
-		}
-
-		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		cert := &certificate.Certificate{}
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
 			corev1Secret, err := b.kubeController.GetSecret(ctx, secret.Namespace, secret.Name)
 			if err != nil {
 				return err
 			}
 
+			bootstrapCert, err := getCertFromSecret(corev1Secret)
+			cert = bootstrapCert
+			if err != nil {
+				log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrObtainingCertFromSecret)).
+					Msgf("Error getting cert %s from bootstrap secret %s/%s", bootstrapCert, corev1Secret.Namespace, corev1Secret.Name)
```

Comment:
Let's only log the CN or serial number of the certificate. In `bootstrapCert` could be nil here, so I think it is sufficient to log "Error getting cert from bootstrap secret %s/%s".